### PR TITLE
CDK: tool for auto-generating schemas from the stream output

### LINKED ISSUE #3712: CDK: Build a tool to populate schemas from the stream outputs
## Tell us about the problem you're trying to solve
Sometimes API docs lack information about data schema. In this case, it is made to just use API response as a base schema.
For this in order to work, we need to ensure that streams have as many kinds of data as possible.

## Describe the solution you’d like
The tool should work as a script that reads output from read command:

```bash
python main.py read --config secrets/config.json --catalog sample_files/configured_catalog.json | schema-generator
```

and should produce files for each stream:

```
stream1.json
stream2.json
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200397516424701) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by vovavovavovavova**
## What
Closes #3712 
## How

`python main.py read --config secrets/config.json --catalog sample_files/configured_catalog.json | python schema_generator.py
`
Command will generate Json schemas and will place them in `_schemas` folder.

### CODE REVIEW

File: tools/integrations/schema_generator.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        schema["properties"] = self._properties_to_schema(self._properties)
+    if self._pattern_properties:
+        schema["patternProperties"] = self._properties_to_schema(self._pattern_properties)
+    return schema
+
+
+Object.to_schema = to_schema
+
+
+data = sys.stdin.readlines()
```
Comment (keu) [inline tools/integrations/schema_generator.py:58]: lets add the main function and call it

File: tools/integrations/schema_generator.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+default_folder = os.path.join(os.getcwd(), "_schemas")
```
Comment (keu) [inline tools/integrations/schema_generator.py:60]: lets just call it `schemas` folder

File: tools/integrations/schema_generator.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        builders[stream_name] = builder
+    else:
+        builder = builders[stream_name]
+    builder.add_object(record.record.data)
+
+for stream_name, builder in builders.items():
+    schema = builder.to_schema()
+    output_file_name = os.path.join(default_folder, stream_name + ".json")
+    with open(output_file_name, "w") as outfile:
+        json.dump(schema, outfile, indent=2)
```
Comment (keu) [inline tools/integrations/schema_generator.py:81]: lets sort keys as well

File: tools/integrations/schema_generator.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+default_folder = os.path.join(os.getcwd(), "_schemas")
+if not os.path.exists(default_folder):
+    os.mkdir(default_folder)
+
+messages = [AirbyteMessage.parse_raw(line) for line in data]
+record_messages = [i for i in messages if i.type == Type.RECORD]
+builders = {}  # "stream_name": builder_instance, "stream_name2: builder_instance2...}
+
+for record in record_messages:
```
Comment (keu) [inline tools/integrations/schema_generator.py:68]: it is matter of taste, but about to iterate over stdin and do processing at the same time?

File: tools/integrations/schema_generator.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    schema = super(Object, self).to_schema()
+    schema["type"] = "object"
+    if self._properties:
+        schema["properties"] = self._properties_to_schema(self._properties)
+    if self._pattern_properties:
+        schema["patternProperties"] = self._properties_to_schema(self._pattern_properties)
+    return schema
+
+
+Object.to_schema = to_schema
```
Comment (keu) [inline tools/integrations/schema_generator.py:55]: no need for hacks, genson is really flexible and quality package, check `EXTRA_STRATEGIES` docs

**Review by keu — CHANGES_REQUESTED** [review]
see my comments

File: tools/integrations/schema_generator.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import json
+import os
+import sys
+
+from airbyte_cdk.models import AirbyteMessage, Type
+from genson import SchemaBuilder
+from genson.schema.strategies.object import Object
+
+
+class NoRequiredObj(Object):
```
Comment (keu) [inline tools/integrations/schema_generator.py:34]: please add class docstring explaining why do we need this

File: tools/integrations/schema_generator.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+def main():
+    data = sys.stdin.readlines()
+
+    default_folder = os.path.join(os.getcwd(), "schemas")
+    if not os.path.exists(default_folder):
+        os.mkdir(default_folder)
+
+    builders = {}
+    for line in data:
```
Comment (keu) [inline tools/integrations/schema_generator.py:57]:
```suggestion
    for line in sys.stdin:
```

**Review by keu — APPROVED** [review]
few comments

**Comment by bazarnov** [comment]
`python main.py read --config secrets/config.json --catalog sample_files/configured_catalog.json | python schema_generator.py`

I have a question, regards this command: 
If I understood this correctly the output would be the same schemas but in separated files that configured_catalog.json is using in order to produce the RECORDS output? 

If so, shouldn't that be something like this? 
`python main.py read --config secrets/config.json | python schema_generator.py >> {output folder with configured_catalog.json + schemas}`

**Pull request merged**